Library/Homebrew/cmd/--cache.rb

```
@@ -30,18 +31,21 @@ def __cache
     if args.no_named?
       puts HOMEBREW_CACHE
     else
-      args.formulae_and_casks.each do |formula_or_cask|
-        case formula_or_cask
-        when Formula
-          formula = formula_or_cask
+      args.named.each do |name|
+        begin
+          formula = Formulary.factory name
           if Fetch.fetch_bottle?(formula)
             puts formula.bottle.cached_download
           else
             puts formula.cached_download
           end
-        when Cask::Cask
-          cask = formula_or_cask
-          puts "cask: #{Cask::Cmd::Cache.cached_location(cask)}"
+        rescue FormulaUnavailableError => e
+          begin
+            cask = Cask::CaskLoader.load name
+            puts "cask: #{Cask::Cmd::Cache.cached_location(cask)}"
+          rescue Cask::CaskUnavailableError
```

If I am reading this correctly, this error message is something generic for either Formula or Cask. In that case we might not want to throw a CaskUnavailableError error? 

One way is to have another `ofail` statement under line 42.

---

I'm not sure where we're throwing a CaskUnavailableError (except possibly `Cask::CaskLoader.load`, but I don't think that's what you mean), and I don't think either of the error messages are meant for both formulae or casks. Do you think that the code should print a generic message when it cannot load a formula or a cask, similar to `--cache`?

---

Ok my bad, I got a little confused with the intention of the program. 

> Do you think that the code should print a generic message when it cannot load a formula or a cask, similar to --cache?

Yes,, that sums it up perfectly. Instead of `CaskUnavailableError`, it should be a more generic error.